src/radical/pilot/configs/resource_princeton.json

```diff
@@ -51,25 +52,26 @@
         "local"                      :
         {
             "job_manager_endpoint"    : "slurm://localhost/",
+            "job_manager_hop"         : "fork://localhost/",
```

Comment:
Out of curiosity, what does `job_manager_hop` do? If it is a placeholder for schema and hostname of `js_url`, empty `job_manager_hop` still has the same behavior? like falling into `elif '+' not in js_url.schema:` condition in `session.py` if it is missing.

Reply:
Ah, good question!

This is somewhat of a historic remnant.  In some cases we had to  use resource URLs to access a batch systems of a cluster via another gateway machine.  In those cases it was difficult to derive the endpoint for file transfer and command submission (where to run `qsub`) from the URL available for the actual resource, and we used the `js_hop` to hardcode that.

By now the code to derive the URL  is more clever, and we handle gateway machines differently, so that setting is not used much - but the code is still there and probably should be cleaned out.

The only really useful setting is for localhost and testing (like here), where I can configure RP to use local `slurm` for submission, but to access the login node via ssh.  That then works even when I am on the loginn node, where the heuristics would think ssh is not needed.  That way we can test ssh (and other remote access mechanisms) when running on localhost.